Spigot-API-Patches/0186-Thrown-projectiles-display-item-methods.patch

```diff
@@ -0,0 +1,90 @@
+@@ -0,0 +1,31 @@
++package com.destroystokyo.paper.entity.projectile;
++
++import org.bukkit.Material;
++import org.bukkit.entity.Projectile;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents a thrown projectile.
++ */
++public interface ThrownProjectile extends Projectile {
++
++    /**
++     * Get this thrown projectile's current display item.
++     *
++     * @return This thrown projectile's current display item.
++     */
++    @NotNull
++    ItemStack getDisplayItem();
++
++    /**
++     * Set this thrown projectile's display item. If the new display item
++     * is null or its type is {@link Material#isEmpty}, the display item
++     * will be set to the default item (snowball if the projectile was
++     * originally a snowball, ender pearl if it was an ender pearl, etc.).
++     *
++     * @param displayItem The new display item.
++     */
++    void setDisplayItem(@Nullable ItemStack displayItem);
++}
+diff --git a/src/main/java/org/bukkit/entity/Egg.java b/src/main/java/org/bukkit/entity/Egg.java
+index 2dcc00b9..6df40f5d 100644
+--- a/src/main/java/org/bukkit/entity/Egg.java
++++ b/src/main/java/org/bukkit/entity/Egg.java
+@@ -3,4 +3,4 @@ package org.bukkit.entity;
+ /**
+  * Represents a thrown egg.
+  */
+-public interface Egg extends Projectile {}
++public interface Egg extends com.destroystokyo.paper.entity.projectile.ThrownProjectile {} // Paper
+diff --git a/src/main/java/org/bukkit/entity/EnderPearl.java b/src/main/java/org/bukkit/entity/EnderPearl.java
+index db18a90b..1b17eb2c 100644
+--- a/src/main/java/org/bukkit/entity/EnderPearl.java
++++ b/src/main/java/org/bukkit/entity/EnderPearl.java
+@@ -3,6 +3,6 @@ package org.bukkit.entity;
+ /**
+  * Represents a thrown Ender Pearl entity
+  */
+-public interface EnderPearl extends Projectile {
++public interface EnderPearl extends com.destroystokyo.paper.entity.projectile.ThrownProjectile { // Paper
+ 
+ }
+diff --git a/src/main/java/org/bukkit/entity/Snowball.java b/src/main/java/org/bukkit/entity/Snowball.java
+index 8c6b4333..c7f80805 100644
+--- a/src/main/java/org/bukkit/entity/Snowball.java
++++ b/src/main/java/org/bukkit/entity/Snowball.java
+@@ -3,4 +3,4 @@ package org.bukkit.entity;
+ /**
+  * Represents a snowball.
+  */
+-public interface Snowball extends Projectile {}
++public interface Snowball extends com.destroystokyo.paper.entity.projectile.ThrownProjectile {} // Paper
+diff --git a/src/main/java/org/bukkit/entity/ThrownExpBottle.java b/src/main/java/org/bukkit/entity/ThrownExpBottle.java
+index 671282ed..2a6e0c7b 100644
+--- a/src/main/java/org/bukkit/entity/ThrownExpBottle.java
++++ b/src/main/java/org/bukkit/entity/ThrownExpBottle.java
+@@ -3,6 +3,6 @@ package org.bukkit.entity;
+ /**
+  * Represents a thrown Experience bottle.
+  */
+-public interface ThrownExpBottle extends Projectile {
++public interface ThrownExpBottle extends Projectile { // Paper
```

Comment:
Did you miss this one, or intentional?

Reply:
I missed this one, will commit a fix in a moment.